searx/engines/command.py

```diff
@@ -0,0 +1,182 @@
+        else:
+            cmd.append(c)
+
+    return cmd
+
+
+def _get_results_from_process(results, process, pageno):
+    leftover = ''
+    count = 0
+    start, end = __get_results_limits(pageno)
+    line = process.stdout.readline()
+    while line:
+        buf = leftover + line.decode('utf-8')
+        raw_results = buf.split(result_separator)
+        if raw_results[-1]:
+            leftover = raw_results[-1]
+        raw_results = raw_results[:-1]
+
+        for raw_result in raw_results:
+            result = __parse_single_result(raw_result)
+            if result is None:
+                _command_logger.debug('skipped result:', raw_result)
+                continue
+
+            if start <= count and count <= end:
+                result['template'] = result_template
+                results.append(result)
+
+            count += 1
+            if end < count:
+                process.stdout.close()
+                return results
+
+        line = process.stdout.readline()
+
+    process.stdout.close()
+
+
+def __get_results_limits(pageno):
+    start = (pageno - 1) * 10
+    end = start + 9
+    return start, end
+
+
+def __check_query_params(params):
+    if not query_type:
+        return
+
+    if query_type == 'path':
+        query_path = params[-1]
+        query_path = expanduser(query_path)
+        if commonprefix([realpath(query_path), working_dir]) != working_dir:
+            raise ValueError('requested path is outside of configured working directory')
+    elif query_type == 'enum' and len(query_enum) > 0:
+        for param in params:
+            if param not in query_enum:
+                raise ValueError('submitted query params is not allowed', param, 'allowed params:', query_enum)
+
+
+def check_parsing_options(engine_settings):
+    """ Checks if delimiter based parsing or regex parsing is configured correctly """
+
+    if 'delimiter' not in engine_settings and 'parse_regex' not in engine_settings:
+        raise ValueError('failed to init settings for parsing lines: missing delimiter or parse_regex')
+    if 'delimiter' in engine_settings and 'parse_regex' in engine_settings:
+        raise ValueError('failed to init settings for parsing lines: too many settings')
+
+    if 'delimiter' in engine_settings:
+        if 'chars' not in engine_settings['delimiter'] or 'keys' not in engine_settings['delimiter']:
+            raise ValueError
+
+
+def __parse_single_result(raw_result):
+    """ Parses command line output based on configuration """
+
+    result = {}
+
+    if delimiter:
+        elements = str_split(raw_result, delimiter['chars'], len(delimiter['keys']) - 1)
+        if len(elements) != len(delimiter['keys']):
+            return {}
+        for i in range(len(elements)):
+            result[delimiter['keys'][i]] = elements[i]
+
+    if parse_regex:
+        for result_key, regex in parse_regex.items():
+            found = re_search(regex, raw_result, flags=MULTILINE)
```

Comment:
If the regex is compiled in the function init(...):
* it gives some speed improvement (not sure here it is relevant)
* if the regex has a syntax error, the developer doesn't have to make a search to see the error.